src/Compilers/Core/Portable/InternalUtilities/OrderedMultiDictionary.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Roslyn.Utilities
+{
+    // Note that this is not threadsafe for concurrent reading and writing.
+    internal sealed class OrderedMultiDictionary<K, V> : IEnumerable<KeyValuePair<K, SetWithInsertionOrder<V>>>
```

Comment:
Consider adding some tests for this type to Compilers/Core/CodeAnalysisTest/InternalUtilities.